module java.base does not export sun.security.tools.keytool to unnamed module @1ece4432

Error happens with latest openjdk 16.0.1.
Works fine with LTS version (openjdk 11.0.11) .


```
Sat Jul 24 09:57:57 PM EEST 2021 Starting heritrix
Linux f34 5.13.4-200.fc34.x86_64 #1 SMP Tue Jul 20 20:27:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
openjdk version "16.0.1" 2021-04-20
OpenJDK Runtime Environment 21.3 (build 16.0.1+9)
OpenJDK 64-Bit Server VM 21.3 (build 16.0.1+9, mixed mode, sharing)
JAVA_OPTS= -Xmx256m
real-time non-blocking time  (microseconds, -R) unlimited
core file size              (blocks, -c) unlimited
data seg size               (kbytes, -d) unlimited
scheduling priority                 (-e) 0
file size                   (blocks, -f) unlimited
pending signals                     (-i) 15505
max locked memory           (kbytes, -l) 64
max memory size             (kbytes, -m) unlimited
open files                          (-n) 1024
pipe size                (512 bytes, -p) 8
POSIX message queues         (bytes, -q) 819200
real-time priority                  (-r) 0
stack size                  (kbytes, -s) 8192
cpu time                   (seconds, -t) unlimited
max user processes                  (-u) 15505
virtual memory              (kbytes, -v) unlimited
file locks                          (-x) unlimited
Red Hat, Inc. OpenJDK Runtime Environment 16.0.1+9
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalAccessException: class org.archive.util.KeyTool cannot access class sun.security.tools.keytool.Main (in module java.base) because module java.base does not export sun.security.tools.keytool to unnamed module @1ece4432
        at org.archive.crawler.Heritrix.useAdhocKeystore(Heritrix.java:459)
        at org.archive.crawler.Heritrix.instanceMain(Heritrix.java:314)
        at org.archive.crawler.Heritrix.main(Heritrix.java:184)
Caused by: java.lang.RuntimeException: java.lang.IllegalAccessException: class org.archive.util.KeyTool cannot access class sun.security.tools.keytool.Main (in module java.base) because module java.base does not export sun.security.tools.keytool to unnamed module @1ece4432
        at org.archive.util.KeyTool.main(KeyTool.java:45)
        at org.archive.crawler.Heritrix.useAdhocKeystore(Heritrix.java:442)
        ... 2 more
Caused by: java.lang.IllegalAccessException: class org.archive.util.KeyTool cannot access class sun.security.tools.keytool.Main (in module java.base) because module java.base does not export sun.security.tools.keytool to unnamed module @1ece4432
        at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:385)
        at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:687)
        at java.base/java.lang.reflect.Method.invoke(Method.java:559)
        at org.archive.util.KeyTool.main(KeyTool.java:40)
        ... 3 more
```

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -10,7 +10,7 @@ jobs:
   build:
     strategy:
       matrix:
-        jdk: [8, 11]
+        jdk: [8, 11, 16]
     runs-on: ubuntu-latest
     timeout-minutes: 30
 

--- a/commons/src/main/java/org/archive/util/KeyTool.java
+++ b/commons/src/main/java/org/archive/util/KeyTool.java
@@ -18,7 +18,13 @@
  */
 package org.archive.util;
 
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 /**
  * Wrapper for "keytool" utility main class. Loads class dynamically, trying
@@ -36,8 +42,20 @@ public class KeyTool {
 				// java 8
 				cl = ClassLoader.getSystemClassLoader().loadClass("sun.security.tools.keytool.Main");
 			}
-			Method main = cl.getMethod("main", new String[0].getClass());
+			Method main = cl.getMethod("main", String[].class);
 			main.invoke(null, (Object) args);
+		} catch (IllegalAccessException e) {
+			// java 16
+			List<String> command = new ArrayList<>();
+			command.add(System.getProperty("java.home") + File.separator + "bin" + File.separator + "keytool");
+			command.addAll(Arrays.asList(args));
+			try {
+				new ProcessBuilder(command).inheritIO().start().waitFor();
+			} catch (IOException e2) {
+				throw new UncheckedIOException(e2);
+			} catch (InterruptedException e2) {
+				Thread.currentThread().interrupt();
+			}
 		} catch (Exception e) {
 			if (e instanceof RuntimeException) {
 				throw (RuntimeException) e;

--- a/modules/pom.xml
+++ b/modules/pom.xml
@@ -31,8 +31,13 @@
 		</dependency>
 		<dependency>
 			<groupId>org.codehaus.groovy</groupId>
-			<artifactId>groovy-all</artifactId>
-			<version>1.6.3</version>
+			<artifactId>groovy-jsr223</artifactId>
+			<version>3.0.8</version>
+		</dependency>
+		<dependency>
+			<groupId>org.codehaus.groovy</groupId>
+			<artifactId>groovy-templates</artifactId>
+			<version>3.0.8</version>
 		</dependency>
 		<dependency>
 			<groupId>org.eclipse.jetty</groupId>

